src/cargo/sources/registry/index.rs

```diff
@@ -266,14 +266,17 @@ impl<'cfg> RegistryIndex<'cfg> {
         &mut self,
         dep: &Dependency,
         load: &mut RegistryData,
+        allow_yanked_deps: bool,
         f: &mut FnMut(Summary),
     ) -> CargoResult<()> {
         let source_id = self.source_id.clone();
         let name = dep.package_name().as_str();
         let summaries = self.summaries(name, load)?;
         let summaries = summaries
             .iter()
-            .filter(|&&(_, yanked)| dep.source_id().precise().is_some() || !yanked)
+            .filter(|&&(_, yanked)| {
+                dep.source_id().precise().is_some() || allow_yanked_deps || !yanked
```

Comment:
I think `self.config` is accessible here, could that be used to avoid passing in `allow_yanked_deps`?

Reply:
Fixed.